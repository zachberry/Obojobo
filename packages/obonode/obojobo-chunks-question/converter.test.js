jest.mock('obojobo-document-engine/src/scripts/common/index', () => ({
	Registry: {
		getItemForType: () => ({
			slateToObo: jest.fn(),
			oboToSlate: jest.fn()
		})
	}
}))

import Converter from './converter'
const SOLUTION_NODE = 'ObojoboDraft.Chunks.Question.Solution'
const MCASSESSMENT_NODE = 'ObojoboDraft.Chunks.MCAssessment'
const NUMERIC_ASSESSMENT_NODE = 'ObojoboDraft.Chunks.NumericAssessment'
const BREAK_NODE = 'ObojoboDraft.Chunks.Break'

describe('Question editor', () => {
	test('slateToObo converts a Slate node to an OboNode with content', () => {
		const slateNode = {
			key: 'mockKey',
			type: 'mockType',
			data: {
				get: () => null
			},
			nodes: [
				{
					type: 'oboeditor.component',
					nodes: [
						{
							type: 'mockNode'
						}
					]
				},
				{
					type: MCASSESSMENT_NODE
				},
				{
					type: SOLUTION_NODE,
					nodes: {
						get: () => ({
							type: 'oboeditor.component',
							nodes: [
								{
									type: 'mockNode'
								}
							]
						})
					}
				}
			]
		}
		const oboNode = Converter.slateToObo(slateNode)

		expect(oboNode).toMatchSnapshot()
	})

	test('oboToSlate converts an OboNode to a Slate node', () => {
		const oboNode = {
			id: 'mockKey',
			type: 'mockType',
			children: [
				{
					type: BREAK_NODE
				},
				{
					type: MCASSESSMENT_NODE
				}
			],
			content: { solution: {} }
		}
		const slateNode = Converter.oboToSlate(oboNode)

		expect(slateNode).toMatchSnapshot()
	})

	test('oboToSlate converts an OboNode to a Slate node without a solution', () => {
		const oboNode = {
			id: 'mockKey',
			type: 'mockType',
			children: [
				{
					type: BREAK_NODE
				},
				{
					type: MCASSESSMENT_NODE
				}
			],
			content: {}
		}
		const slateNode = Converter.oboToSlate(oboNode)

		expect(slateNode).toMatchSnapshot()
	})

	test('slateToObo converts a Slate node to an OboNode for NumericAssessment', () => {
		const slateNode = {
			key: 'mockKey',
			type: 'mockType',
			data: {
				get: () => null
			},
			nodes: [
				{
					type: 'oboeditor.component',
					nodes: [
						{
							type: 'mockNode'
						}
					]
				},
				{
					type: NUMERIC_ASSESSMENT_NODE
				},
				{
					type: SOLUTION_NODE,
					nodes: {
						get: () => ({
							type: 'oboeditor.component',
							nodes: [
								{
									type: 'mockNode'
								}
							]
						})
					}
				}
			]
		}
		const oboNode = Converter.slateToObo(slateNode)

		expect(oboNode).toMatchSnapshot()
	})
})
