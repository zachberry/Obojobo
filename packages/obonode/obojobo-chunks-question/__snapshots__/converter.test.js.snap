// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question editor oboToSlate converts an OboNode to a Slate node 1`] = `
Object {
  "data": Object {
    "content": Object {
      "solution": Object {},
    },
  },
  "key": "mockKey",
  "nodes": Array [
    Object {
      "nodes": Array [
        undefined,
      ],
      "object": "block",
      "type": "oboeditor.component",
    },
    undefined,
    Object {
      "nodes": Array [
        undefined,
      ],
      "object": "block",
      "type": "ObojoboDraft.Chunks.Question.Solution",
    },
  ],
  "object": "block",
  "type": "mockType",
}
`;

exports[`Question editor oboToSlate converts an OboNode to a Slate node without a solution 1`] = `
Object {
  "data": Object {
    "content": Object {},
  },
  "key": "mockKey",
  "nodes": Array [
    Object {
      "nodes": Array [
        undefined,
      ],
      "object": "block",
      "type": "oboeditor.component",
    },
    undefined,
  ],
  "object": "block",
  "type": "mockType",
}
`;

exports[`Question editor slateToObo converts a Slate node to an OboNode for NumericAssessment 1`] = `
Object {
  "children": Array [
    undefined,
    undefined,
  ],
  "content": Object {
    "solution": undefined,
  },
  "id": "mockKey",
  "type": "mockType",
}
`;

exports[`Question editor slateToObo converts a Slate node to an OboNode with content 1`] = `
Object {
  "children": Array [
    undefined,
    undefined,
  ],
  "content": Object {
    "solution": undefined,
  },
  "id": "mockKey",
  "type": "mockType",
}
`;
